linode_api4/groups/database.py

```diff
@@ -126,13 +127,71 @@ def mysql_create(self, label, region, engine, ltype, **kwargs):
 
         params = {
             "label": label,
-            "region": region.id if issubclass(type(region), Base) else region,
-            "engine": engine.id if issubclass(type(engine), Base) else engine,
-            "type": ltype.id if issubclass(type(ltype), Base) else ltype,
+            "region": region,
+            "engine": engine,
+            "type": ltype,
         }
         params.update(kwargs)
 
-        result = self.client.post("/databases/mysql/instances", data=params)
+        result = self.client.post(
+            "/databases/mysql/instances",
+            data=_flatten_request_body_recursive(drop_null_keys(params)),
+        )
+
+        if "id" not in result:
+            raise UnexpectedResponseError(
+                "Unexpected response when creating MySQL Database", json=result
+            )
+
+        d = MySQLDatabase(self.client, result["id"], result)
+        return d
+
+    def mysql_fork(self, source, restore_time, **kwargs):
+        """
+        Forks an :any:`MySQLDatabase` on this account with
+        the given restore_time. label, region, engine, and ltype are optional.
+        For example::
+
+           client = LinodeClient(TOKEN)
+
+           db_to_fork = client.database.mysql_instances()[0]
+
+           new_fork = client.database.mysql_fork(
+                db_to_fork.id,
+                db_to_fork.updated,
+                label="new-fresh-label"
+           )
+
+        API Documentation: https://techdocs.akamai.com/linode-api/reference/post-databases-mysql-instances
+
+        :param source: The id of the source database
+        :type source: int
+        :param restore_time: The timestamp for the fork
+        :type source: datetime
+        :param label: The name for this cluster
+        :type label: str
+        :param region: The region to deploy this cluster in
+        :type region: str or Region
```

Comment:
I think we can use Union or `|` operator to express "or"?
```suggestion
        :type region: str | Region
```